### ISSUE TITLE
Need ability to disable IPC usage for platforms beyond Android

### ISSUE THREAD

**Issue by ykku16github** [intel/media-driver#854]
The following 2 files disable IPC usage for the Android platform:
https://github.com/intel/media-driver/blob/master/media_driver/linux/common/os/mos_context_specific.cpp
https://github.com/intel/media-driver/blob/master/media_driver/linux/common/os/mos_os_specific.c

There is a need to do the same thing on other platforms, such as Chrome OS.  

It would be nice if the IPC-related code is encapsulated by a slightly more meaningful name.  For example, the name could describe the feature being supported.  Chrome OS or any other platform can then switch it on or off when calling cmake.  This suggestion is credited to Miguel Casas <mcasas@chromium.org>.

**Comment by XinfengZhang** [comment]
good idea, the IPC is used for gen9 task balance, we could use random dispatch to replace it.

**Comment by paulmenzel** [comment]
What is the status of this? [It’s also relevant for using the Intel driver with Mozilla Firefox.](https://bugzilla.mozilla.org/show_bug.cgi?id=1619585#c46)

**Comment by Darkspirit** [comment]
Regarding Firefox it should be noted that - still experimental - VAAPI decoding will be moved from the sandboxed content process to the unsandboxed GPU process. Current sandbox violations don't seem to matter then.

**Comment by XinfengZhang** [comment]
actually, for Iris Xe, we do not need IPC any more, the IPC call is only for Gen9 (SKL, KBL,...) which has no virtual engine support. 

We already have a patch which remove the IPC call , but it is not merged because of regressions and concerns. 
for gen9 platforms like SKL.   it has two VDBoxes, so, to balance the task from different process, we need IPC call to reserve the task information on these two VDBoxes, then choose the "less busy" VDBox to dispatch new task. 
after change the policy to random, it will be "balanced" if the process number is enough, but if there are just 2 or 3 process, it is possible that all task will be dispatched to 1 VDbox...

**Comment by ravenexp** [comment]
CPU: Core i5-1135G7
DEV ID: `0x9a49`

**Comment by paulmenzel** [comment]
@XinfengZhang, thank you for working on this issue. As this issues seems to [prevent the Firefox developers to enable VA-API video decoding by default in Mozilla Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1698778), do you have a time frame, when all generations will be fixed?

**Comment by heftig** [comment]
It seems ICL is not fixed yet, either.

**Comment by XinfengZhang** [comment]
#1293

**Comment by eero-t** [comment]
> #1293 could be used to fix ICL, SKL...

@XinfengZhang Are the other platforms fixed by now i.e. could this be closed?
